### PR TITLE
Add support for automanage-vhd host targets

### PR DESCRIPTION
**Pull request by jterry75**
Signed-off-by: Justin Terry (VM) <juterry@microsoft.com>

### CODE REVIEW

**Comment by jterry75** [comment]
@abwah - FYI

**Comment by jterry75** [comment]
Btw this fixes some bugs I found as well. If you want a separate PR for that I would be happy to separate them out.

**Comment by jterry75** [comment]
__Don't merge this yet. Need to finish testing__

File: pkg/server/container_create_windows.go [hunk 1]
```diff
@@ -318,11 +319,46 @@ func (c *criService) generateContainerSpec(id string, sandboxID string, sandboxP
...
+			s, err := c.os.Stat(formattedSource)
+			if err != nil {
+				return nil, errors.Wrapf(err, "failed to Stat automanage-vhd:// mount.HostPath %q", formattedSource)
+			}
+			if s.IsDir() {
+				// TODO: JTERRY75 - This is a hack and needs to be removed. The
+				// orchestrator should be controlling the entry host path. For
+				// now if its a directory copy the template vhd and update the
+				// source.
+				if c.config.PluginConfig.TmpTemplateVhdPath == "" {
```
Comment (kevpar) [inline pkg/server/container_create_windows.go]: As this is generalized beyond just the /tmp case, we should rename the config value to something like `AutoManageVHDTemplatePath`.

Comment (jterry75) [reply pkg/server/container_create_windows.go]: Deal

File: pkg/config/config.go [hunk 2]
```diff
@@ -151,11 +151,12 @@ type PluginConfig struct {
 	// current OOMScoreADj.
 	// This is useful when the containerd does not have permission to decrease OOMScoreAdj.
 	RestrictOOMScoreAdj bool `toml:"restrict_oom_score_adj" json:"restrictOOMScoreAdj"`
-	// TmpTemplateVhdPath is the path on disk to find the source (template) vhdx to
-	// copy when a user specifies a /tmp override with a source directory.
+	// AutoManageVHDTemplatePath is the path on disk to find the source
+	// (template) vhd to copy when a user specifies a mount with a `HostPath:
+	// "automanage-vhd://"` prefix were the `HostPath` is a directory.
```
Comment (kevpar) [inline pkg/config/config.go]: "where"

**Pull request merged**